build.gradle.kts

```diff
@@ -104,6 +104,13 @@ dependencyAnalysis {
         exclude("com.bucket4j:bucket4j-core")
       }
     }
+    project(":misk-action-scopes") {
+      onIncorrectConfiguration {
```

Comment:
I haven't encountered this Gradle block before, how does it work? If it excludes `misk-api` does that mean sometimes we're silently failing configuration/tests for that project?

Reply:
There is an existing one here
https://github.com/cashapp/misk/pull/3115/files#diff-c0dfa6bc7a8685217f70a860145fbdf416d449eaff052fa28352c5cec1a98c06L86

The error I was getting from the dependency-analysis plugin before adding this block was:
```
> Existing dependencies which should be modified to be as indicated:
    implementation(project(":misk-api")) (was api)
```

Essentially saying no misk-api classes are used in action-scope's public APIs. But for backwards compatibility, I want to have this an an API dependency (so that any projects depending on misk-action-scope and using the types moved to misk-api still compile).